files/grest/rpc/01_cached_tables/stake_distribution_cache.sql

```diff
@@ -1,301 +1,238 @@
-    max(withdrawal.tx_id) as latest_withdrawal_tx_id
-  FROM WITHDRAWAL
-    INNER JOIN accounts_with_delegated_pools ON accounts_with_delegated_pools.stake_address_id = WITHDRAWAL.addr_id
-  GROUP BY accounts_with_delegated_pools.stake_address_id
-);
--- possibly should remove this as a TEMP table selecting a column that is indexed will carry the index over
-CREATE INDEX IF NOT EXISTS idx_stake_address_id ON account_last_withdrawal (stake_address_id);
-/* 
- * TABLE: account_total_reserves
- * stake_addresss_id
- * RESERVES (total reserves value for each account - excluding already withdrawn funds, which are accounted for in the UTXO)
- */
-CREATE TEMP TABLE account_total_reserves as (
-  SELECT accounts_with_delegated_pools.stake_address_id,
-    COALESCE(SUM(RESERVE.AMOUNT), 0) AS RESERVES
-  FROM RESERVE
-    INNER JOIN accounts_with_delegated_pools ON accounts_with_delegated_pools.stake_address_id = RESERVE.addr_id
-    LEFT JOIN account_last_withdrawal ON account_last_withdrawal.stake_address_id = reserve.addr_id
-  WHERE RESERVE.tx_id >= COALESCE(
-      account_last_withdrawal.latest_withdrawal_tx_id,
-      0
-    )
-  GROUP BY accounts_with_delegated_pools.stake_address_id
-);
--- possibly should remove this as a TEMP table selecting a column that is indexed will carry the index over
-CREATE INDEX IF NOT EXISTS idx_stake_address_id ON account_total_reserves (stake_address_id);
-/* 
- * TABLE: account_total_treasury
- * stake_addresss_id
- * TREASURY (total treasury value for each account - excluding already withdrawn funds, which are accounted for in the UTXO)
- */
-CREATE TEMP TABLE account_total_treasury as (
-  SELECT accounts_with_delegated_pools.stake_address_id,
-    COALESCE(SUM(TREASURY.AMOUNT), 0) AS TREASURY
-  FROM TREASURY
-    INNER JOIN accounts_with_delegated_pools ON accounts_with_delegated_pools.stake_address_id = TREASURY.addr_id
-    LEFT JOIN account_last_withdrawal ON account_last_withdrawal.stake_address_id = accounts_with_delegated_pools.stake_address_id
-  WHERE TREASURY.tx_id >= COALESCE(
-      account_last_withdrawal.latest_withdrawal_tx_id,
-      0
-    )
-  GROUP BY accounts_with_delegated_pools.stake_address_id
-);
--- possibly should remove this as a TEMP table selecting a column that is indexed will carry the index over
-CREATE INDEX IF NOT EXISTS idx_stake_address_id ON account_total_treasury (stake_address_id);
-/* 
- * INSERT QUERY START
- */
-INSERT INTO GREST.STAKE_DISTRIBUTION_CACHE
-SELECT awdp.stake_address,
-  pi.pool_id,
-  CASE
-    WHEN (
-      COALESCE(atrew.REWARDS, 0) - COALESCE(atw.WITHDRAWALS, 0)
-    ) < 0 THEN COALESCE(atu.UTXO, 0) + COALESCE(atrew.REWARDS, 0) - COALESCE(atw.WITHDRAWALS, 0) + COALESCE(atres.RESERVES, 0) + COALESCE(att.TREASURY, 0) - (
-      COALESCE(atrew.REWARDS, 0) - COALESCE(atw.WITHDRAWALS, 0)
+  SELECT block_no FROM PUBLIC.BLOCK
+    WHERE block_no IS NOT NULL
+      AND block_no = (SELECT MAX(BLOCK_NO) FROM PUBLIC.BLOCK) INTO _last_accounted_block_height;
```

Comment:
Is `MAX(BLOCK_NO)` correct here? We may want to look 5 blocks behind the tip?

Reply:
Not necessarily, even if block is rolled back and tx is included elsewhere - it would prolly end up to be same value